### DISCUSSION TITLE
Extended knot sequence for degree > 1

### DISCUSSION THREAD

**Discussion by matthiaskaeding** [wenjie2wang/splines2#8] [Q&A] [answered]
Hi,

there is something I do not understand about splines2::bSpline, I tried to get it from the code but couldn`t find it:

If one sets the degree to a value >= 1, additional knots are required. How are they chosen, and is it possible to adjust those?

Cheers
Matthias

**Comment by wenjie2wang** [comment, marked as answer]
Hi,

> If one sets the degree to a value >= 1, additional knots are required. How are they chosen, and is it possible to adjust those?

Do you mean the additional knots placed at the boundary?  A simple knot sequence is used internally, where each boundary knot is repeated *k+1* times, *k* represents the degree.

An extended knot sequence can be specified via the Rcpp interface.  Or you may want to use `splines::splineDesign()` instead.

**Reply by matthiaskaeding** [reply]
Thank, yes I meant the additional knots at the boundary. I think I will use the Rcpp interface then.
